### ISSUE TITLE
Don't allow unpacking non-iterable

### ISSUE THREAD

**Issue by AndrolGenhald** [vimeo/psalm#5453]
This should be an InvalidArgument: https://psalm.dev/r/410b622a1f
It's currently allowed because the type _can_ be an iterable, but it should be disallowed if the type can be anything other than iterable.

**Comment by weirdan** [comment]
> This should be an `InvalidArgument`

More precisely, it should be `PossiblyInvalidArgument`.

**Comment by AndrolGenhald** [comment]
I'm not sure I fully understand the difference between `InvalidArgument` and `PossiblyInvalidArgument`.

**Comment by weirdan** [comment]
> Should psalm always emit PossiblyInvalidArgument when it's possible the argument has the correct type?

Yes, if an argument type has an intersection with the required parameter type then it's a possibly invalid call: https://psalm.dev/r/686b176d87

`InvalidScalarArgument`, however, currently doesn't have it's `Possibly*` counterpart, so this can be a bit confusing.
